### PR TITLE
Sanitize the pattern before validation for failing K8s components

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/pattern/stages/validate.go [hunk 1]
```diff
@@ -28,7 +29,9 @@ func Validator(prov ServiceInfoProvider, act ServiceActionProvider) ChainStageFu
 				act.Terminate(fmt.Errorf("invalid workload of type: %s", svc.Type))
 				return
 			}
-
+			if k8s.Format {
+				k8s.Format.DePrettify(svc.Settings)
```
Comment (leecalcote) [inline models/pattern/stages/validate.go:33]: Is this only applicable for Services?

**Review by leecalcote — APPROVED** [review]

File: models/pattern/stages/validate.go (hunk 1 above)
Comment (leecalcote) [reply models/pattern/stages/validate.go:33]: Or is this applied generally?

Comment (leecalcote) [reply models/pattern/stages/validate.go:33]: I presume that "Service" in this case is Pattern "Service" not Kubernetes "Service", is that right?

Comment (Revolyssup) [reply models/pattern/stages/validate.go:33]: Yes to your last statement.

Comment (leecalcote) [reply models/pattern/stages/validate.go:33]: This makes much sense. Thanks for confirming.

**Review by leecalcote — APPROVED** [review]

**Pull request merged**